zanata-common-api/src/main/java/org/zanata/common/DocumentType.java

```diff
@@ -83,9 +125,28 @@ public static DocumentType typeFor(String extension) {
     DocumentType(@Nonnull String... extensions) throws IllegalArgumentException {
         if (extensions.length == 0) {
             throw new IllegalArgumentException(
+                "DocumentType must be constructed with at least one extension.");
+        }
+        Map<String, String> extensionsMap = new HashMap<>();
+        for(String extension : extensions) {
+            extensionsMap.put(extension, extension);
+        }
+        this.extensions = unmodifiableMap(extensionsMap);
+    }
+
+    /**
+     * Create a document type enum constant with the given list of extensions.
+     * At least one extension must be specified.
+     *
+     * @throws IllegalArgumentException
+     *             if no extensions are specified
+     */
+    DocumentType(@Nonnull Map<String, String> extensions) throws IllegalArgumentException {
+        if (extensions.isEmpty()) {
```

Comment:
We should definitely keep the check, I was just thinking that I missed a different check when I wrote the code the first time so that we can't accidentally add an empty extension (since that might make the other code in here behave in an unexpected way).